common/leica_sep.c

```diff
@@ -66,23 +70,89 @@ static int leica_sep_transfer(struct leica_sep_funcs *f, char *cmd, char *data,
 	return bytes_received;
 }
 
-int leica_sep_get_board_id(struct leica_sep_funcs *f, struct leica_sep_board_id *board_id)
+/*
+ * leica_get_sep_payload
+ *
+ * Function that takes as argument the received data from the MCU and extracts
+ * the payload. The result is stored overwriting the received argument with a
+ * new string.
+ */
+static int leica_get_sep_payload(char *data, size_t size)
```

Comment:
Since we are only interested in the payload:
Why not restart filling up `data` after `command_received` is set to 1 in `leica_sep_transfer`? And filter `"` of course...
Then we can drop this function at all

Reply:
@htho-lgs at this point I would call `leica_get_sep_payload` from `leica_sep_transfer`. It is better to keep the complexity split in 2 functions